ndb_api/db_helpers.go

```diff
@@ -304,6 +305,202 @@ func (a *PostgresRequestAppender) appendProvisioningRequest(req *DatabaseProvisi
+			primaryNodeCount += 1
+		}
+
+		props := make([]map[string]string, 4)
+		props[0] = map[string]string{
+			"name":  "role",
+			"value": currentNode.Properties.Role,
+		}
+		props[1] = map[string]string{
+			"name":  "failover_mode",
+			"value": currentNode.Properties.FailoverMode,
+		}
+		props[2] = map[string]string{
+			"name":  "node_type",
+			"value": currentNode.Properties.NodeType,
+		}
+		props[3] = map[string]string{
+			"name":  "remove_archive_destination",
+			"value": database.GetAdditionalArguments()["remove_archive_destination"],
+		}
+		req.Nodes = append(req.Nodes, Node{
+			Properties:       props,
+			VmName:           currentNode.VmName,
+			NxClusterId:      database.GetClusterId(),
+			NetworkProfileId: req.NetworkProfileId,
+			ComputeProfileId: req.ComputeProfileId,
+		})
+	}
+	const MinReqDatabaseNodes = 3
+	if nodeCount < MinReqDatabaseNodes {
+		return fmt.Errorf("invalid node count: HA instance needs at least %d nodes, given: %d", MinReqDatabaseNodes, nodeCount)
+	}
+
+	return nil
+}
+
+func createDefaultNodes(database DatabaseInterface) []*v1alpha1.Node {
+	nodes := make([]*v1alpha1.Node, 0)
+	nodes = append(nodes, &v1alpha1.Node{
+		VmName: database.GetAdditionalArguments()["cluster_name"] + "_haproxy1",
+		Properties: v1alpha1.NodeProperties{
+			NodeType: "haproxy",
+		},
+	})
+	nodes = append(nodes, &v1alpha1.Node{
+		VmName: database.GetAdditionalArguments()["cluster_name"] + "_haproxy2",
+		Properties: v1alpha1.NodeProperties{
+			NodeType: "haproxy",
+		},
+	})
+	nodes = append(nodes, &v1alpha1.Node{
+		VmName: database.GetAdditionalArguments()["cluster_name"] + "-1",
+		Properties: v1alpha1.NodeProperties{
+			NodeType:     "database",
+			Role:         "Primary",
+			FailoverMode: "Automatic",
+		},
+	})
+	nodes = append(nodes, &v1alpha1.Node{
+		VmName: database.GetAdditionalArguments()["cluster_name"] + "-2",
+		Properties: v1alpha1.NodeProperties{
+			NodeType:     "database",
+			Role:         "Secondary",
+			FailoverMode: "Automatic",
+		},
+	})
+	nodes = append(nodes, &v1alpha1.Node{
+		VmName: database.GetAdditionalArguments()["cluster_name"] + "-3",
+		Properties: v1alpha1.NodeProperties{
+			NodeType:     "database",
+			Role:         "Secondary",
+			FailoverMode: "Automatic",
+		},
+	})
+	return nodes
+}
+
+func getPrimaryNodeCount(nodesRequested []*v1alpha1.Node) int {
+	count := 0
+	for _, node := range nodesRequested {
+		if node.Properties.Role == "Primary" {
+			count++
+		}
+	}
+	return count
+}
+
+func (a *PostgresHARequestAppender) appendProvisioningRequest(req *DatabaseProvisionRequest, database DatabaseInterface, reqData map[string]interface{}) (*DatabaseProvisionRequest, error) {
+	dbPassword := reqData[common.NDB_PARAM_PASSWORD].(string)
+	databaseNames := database.GetInstanceDatabaseNames()
+	req.SSHPublicKey = reqData[common.NDB_PARAM_SSH_PUBLIC_KEY].(string)
+	// Set the number of nodes to 5, 3 Postgres nodes + 2 HA Proxy nodes
+	err := setNodesParameters(req, database)
```

Comment:
rename variable to `overriteNodeParmaetersForHA`

Reply:
I am not sure of a better name as in this case the node parameters are always being set using this method not just in the case of overriding the defaults. If using the defaults the call to that method to set the defaults happens inside the setNodesParameters method